Cesium3DTilesSelection/include/Cesium3DTilesSelection/Tileset.h

```diff
@@ -409,6 +428,7 @@ class CESIUM3DTILESSELECTION_API Tileset final {
 
   TraversalDetails _renderLeaf(
       const FrameState& frameState,
+      ImplicitTraversalInfo&& implicitInfo,
```

Comment:
IIRC, it can't be a const reference because `pParentNode`  needs to be mutable when adding a new subtree node during subtree load. It made sense to me to move the `implicitInfo` into the `SubtreeLoadRecord` so that `pParentNode`, `usingImplicitQuadtreeTiling`, `usingImplicitOctreeTiling`, were all available to be potentially used in `loadSubtreeContent` without a copy. That being said I definitely failed to create a move constructor for it, which I will do right away unless you still find move semantics unnecessary here.

Reply:
Hm okay, I think I now see what you must have actually meant by "...doesn't benefit from move semantics anyway." `ImplicitTraversalInfo` is pretty much a POD struct since it is not responsible for managing the memory of the pointers. There is no point to move semantics since all of the members are primitive types which will be copied anyways. And in general the struct is not heavy enough to warrant wrapping it in a unique_ptr since it would need to be heap allocated in that case. Hopefully I'm understanding that right, I'm still trying to wrap my head around move semantics.

Reply:
> it can't be a const reference because pParentNode needs to be mutable when adding a new subtree node during subtree load.

You mean in `addSubtreeToLoadQueue`? You just need to copy it and modify the copy. It's a very inexpensive operation for this struct.

> Hopefully I'm understanding that right

Yep, exactly right.

Reply:
I've changed it so it's just doing the copy there (it's a non-const reference now). But if the `ImplicitTraversalInfo` is a const reference instead, then even after copying you could only access a const `info.pParentNode` pointer right? I think we need a non-const `pParentNode` pointer so that we can add a new child subtree node onto it when we begin loading. 

Just so I'm not making a C++ mistake here, if you had this type of situation:

```
struct A {
  Obj* ptr
};

const A a;
```
What is the type of `a.ptr`, is it `const Obj*` or `Obj const*`?

Reply:
Somewhat surprisingly, `a.ptr` will be `Obj* const`, meaning you can't change the value of the pointer, but you _can_ change the thing pointed to by it. See https://godbolt.org/z/WW1qh6rq3

(`Obj const*` and `const Obj*` are identical I believe, the former is just a less commonly used way of expressing it)